### PR TITLE
Surface previous offset when failing events by tag query

### LINKED ISSUE #478: Surface UUID of previous event when EventByTag stream fails due to missing event
When an event by tag query sees a missing event for a persistence id it searches the current and previous time bucket before failing the stream. If the missing event wasn't recent and was missed due to eventual consistency delay being too low or the event only appearing in the database much later due to a node crash and the event being written on the next startup of the persistenceId then the event may be in the database but not found by this search.  

There's already a ticket for making this search go further back (it would be very inefficient as the schema doesn't allow filtering on persistenceId), another solution would be to fail the stream with the previous offset in the exception meaning that if your downstream processing is idempotent the stream could be restarted from that offset.

### PR DESCRIPTION
**Pull request by chbatey**
Closes #478

### CODE REVIEW

File: EventsByTag.md [hunk 1]
```diff
@@ -68,13 +68,22 @@ Order of the events is maintained per persistence id per tag by maintaining a se
...
 delayed events in live queries.
 
-### Gap detection for the same persistence id
+### Gap detection
 
 If a gap is detected then the event is held back and the stream goes into a searching mode for the missing
-events. If the missing event is not found then then stream is failed.
+events. The search only looks in the current bucket and the previous one. If the event is older then it won't be found in this search.
+
+If the missing event is not found then then stream is failed unless it is the first time the persistence id
```
Comment (2m) [inline EventsByTag.md]:
```suggestion
If the missing event is not found then the stream is failed unless it is the first time the persistence id
```

**Comment by chbatey** [comment]
Failed with:

```
[warn] 	[FAILED     ] org.eclipse.jgit#org.eclipse.jgit;4.5.0.201609210915-r!org.eclipse.jgit.jar: null (249ms)
[warn] 	Detected merged artifact: [FAILED     ] org.eclipse.jgit#org.eclipse.jgit;4.5.0.201609210915-r!org.eclipse.jgit.jar: null (249ms).
[warn] ==== public: tried
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::              FAILED DOWNLOADS            ::
[warn] 	:: ^ see resolution messages for details  ^ ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.eclipse.jgit#org.eclipse.jgit;4.5.0.201609210915-r!org.eclipse.jgit.jar
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[error] sbt.librarymanagement.ResolveException: download failed: org.eclipse.jgit#org.eclipse.jgit;4.5.0.201609210915-r!org.eclipse.jgit.jar
[error] 	at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:332)

```

**Comment by 2m** [comment]
Should this be backported to `0.x` branch, @chbatey ?

**Comment by chbatey** [comment]
Let’s wait for someone to request it

**Pull request merged**